### PR TITLE
Dont send undefined to api

### PR DESCRIPTION
**Pull request by jakubsy**

### CODE REVIEW

File: packages/labextension/src/connection.ts [hunk 1]
```diff
@@ -78,9 +78,13 @@ export class NeptuneConnection {
   };
 
   getNotebook = () => {
+
     return this
       .getAuthorizationHeader()
-      .then(authorizationHeader => {
+      .then((authorizationHeader) => {
+        if (!this.params.notebookId) {
```
Comment (kierson) [inline packages/labextension/src/connection.ts]: I think you can return `return Promise.resolve({} as INeptuneNotebook)` at the begining of `getNotebook` method. I don't think we need auth header to determine that we can't get anything useful from API.

File: packages/labextension/src/connection.ts [hunk 2]
```diff
@@ -78,9 +78,13 @@ export class NeptuneConnection {
...
+          return {} as INeptuneNotebook;
```
Comment (kierson) [inline packages/labextension/src/connection.ts]: Frankly I think that returning empty object and saying to TypeScript that it is INeptuneNotebook (it does not have any optional field!) is a hack. However I see that other parts of the code are written in the same manner and fixing this small bug in a proper way would force you to quite big refactor of whole extension.

I think we can agree to merge as is as this is bug fix.
However when will add new features to the extension, we should consider refactoring.

File: packages/labextension/src/connection.ts (hunk 1 above)
Comment (jakubsy) [reply packages/labextension/src/connection.ts]: Fixed.

File: packages/labextension/src/connection.ts (hunk 2 above)
Comment (jakubsy) [reply packages/labextension/src/connection.ts]: Agreed

**Pull request merged**